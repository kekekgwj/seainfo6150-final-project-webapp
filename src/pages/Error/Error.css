

.errorpPage-wrap {
	width: 100%;
	height: 100%;
    margin: 0;
	padding: 0;
	font-family: 'Times New Roman', Times, serif;
	color: #34495e;
}

.errorpPage-box {
	width: 800px;
	height: 400px;
	position: absolute;
	top: 50%;
	margin-top: -200px;
	left: 50%;
	margin-left: -400px;
}

.errorpPage-tip h3 {
	font-size: 24px;
	font-weight: 300;
	margin-top: 20px;
	margin-bottom: 10px;
}

.errorpPage-tip p {
	margin: 20px 0;
}

.errorpPage-operate a {
	margin-left: 10px;
	color: #06F;
	text-decoration: none;
}

.errorpPage-operate a:first-child {
	margin-left: 0 !important;
}

.errorpPage-img,.errorpPage-tip,.errorpPage-operate {
	text-align: center;
}