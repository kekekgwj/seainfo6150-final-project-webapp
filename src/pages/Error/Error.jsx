import './Error.css';
import errorImg from "../../assets/404.jpg";
import { Link } from "react-router-dom";

const Error = ({errorMessage}) => {
    return (
    <div className="errorpPage-wrap">
    <div className="errorpPage-box">
        <div className="errorpPage-img">
            <img src={errorImg} alt="error"/>
        </div>
        <div className="errorpPage-tip">
            <h3>Whoops... Looks like nothing is here!</h3>
            {errorMessage}
        </div>
        <div className="errorpPage-operate">
            

            <Link to="/">
                <span className="glyphicon glyphicon-home"></span>Back to home page
            </Link>
        </div>

    </div>
</div>
)
}

export default Error;