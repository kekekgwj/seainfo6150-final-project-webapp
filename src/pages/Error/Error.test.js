import React from "react";
import Error from "./Error";

describe("Error page tests", () => {
  it("renders correctly", () => {
    const { container } = render(<Error errorMessage="test error message" />);
    expect(container).toMatchSnapshot();
  });
});