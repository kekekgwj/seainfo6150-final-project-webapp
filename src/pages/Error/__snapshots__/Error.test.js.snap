// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error page tests renders correctly 1`] = `
<div>
  <div
    class="errorpPage-wrap"
  >
    <div
      class="errorpPage-box"
    >
      <div
        class="errorpPage-img"
      >
        <img
          alt="error"
          src="404.jpg"
        />
      </div>
      <div
        class="errorpPage-tip"
      >
        <h3>
          Whoops... Looks like nothing is here!
        </h3>
        test error message
      </div>
      <div
        class="errorpPage-operate"
      >
        <a
          href="/"
        >
          <span
            class="glyphicon glyphicon-home"
          />
          Back to home page
        </a>
      </div>
    </div>
  </div>
</div>
`;
